### ISSUE TITLE
how to close the ws on component unmount?

### ISSUE THREAD

**Issue by thelonewolf123** [robtaussig/react-use-websocket#174]
if the share option is set to true, the ws connection is not terminated if I move from one page to another page. I want to manually clean up the ws connections before the component unmounts. Is it possible to do that with `useWebSocket`?

I couldn't find much info on this topic on docs or codebase.

**Comment by robtaussig** [comment]
Hi @thelonewolf123 , apologies for the late reply.

If the `share` option is used, the ws will only close once all subscribers have unsubscribed (either by unmounting or passing `false` as the third paramater to the hook). It's hard for me to suggest a solution to your problem without seeing your code or knowing more about your use-case, but it sounds to me like you probably shouldn't be using the `share` option. If not set (or set explicitly to `false`), the websocket will be closed and a new one opened when changing pages. If your goal is just for the websocket to close(and not open again), you can do something like this:

```
const location = useLocation();
const { lastMessage } = useWebSocket('ws://some-url.com', {}, location.pathname === '/the-page-i-want-the-websocket-to-work-from');
```

It will only when you are on the path that you want, and when you navigate to another path, it will close.

**Comment by thelonewolf123** [comment]
Hi @robtaussig, Thanks for the response

I'm working on an online IDE project. I wanted to implement a custom reconnection logic if the socket is terminated due to a network issue or some other factor. I tried to find ways to simulate connection termination, but it's not possible to terminate a ws connection from dev tools. Finally, I ended up setting a server-side event on the websocket, if I send 'terminate' from the client, server will terminate that connection. I needed this only for development purposes

my issue is solved, I'll close this
